src/Core/Manifest/VersionProvider.php

```diff
@@ -25,13 +25,22 @@
  */
 class VersionProvider
 {
+
     use Configurable;
 
     /**
      * @var array
      */
     private static $modules = [];
 
+    /**
+     * Marked as internal because this is only intended to be used for unit testing
```

Comment:
This is an anti-pattern we've been trying to move away from. Maybe instead you could add a protected `getComposerLockPath(): string` method which you can then mock in unit tests.